src/cmap/connection_pool.ts

```diff
@@ -231,7 +231,7 @@ export class ConnectionPool extends EventEmitter {
     }
 
     this[kWaitQueue].push(waitQueueMember);
-    setImmediate(() => processWaitQueue(this));
+    process.nextTick(() => processWaitQueue(this));
```

Comment:
`process.nextTick()` forwards the second and subsequent arguments to the callback function. It permits to avoid creating an unnecessary closure and improves performance further. It's a very popular pattern across  the Node.js core codebase (https://github.com/nodejs/node/search?p=1&q=process.nextTick+path%3Alib%2F).
```suggestion
    process.nextTick(processWaitQueue, this);
```

Reply:
Thanks for taking the time to suggest this, I agree we can drop the closure